test/test_framework.py

```diff
@@ -774,12 +774,48 @@ def create_framework(self, cls=Framework):
         return framework
 
     def test_basic_state_storage(self):
-        framework = self.create_framework()
+        class SomeObject(Object):
+            state = StoredState()
+
+        self._stored_state_tests(SomeObject)
+
+    def test_straight_subclass(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+        class Sub(SomeObject):
+            pass
+
+        self._stored_state_tests(Sub)
 
+    def test_straight_sub_subclass(self):
         class SomeObject(Object):
             state = StoredState()
 
-        obj = SomeObject(framework, "1")
+        class Sub(SomeObject):
+            pass
+
+        class SubSub(SomeObject):
+            pass
+
+        self._stored_state_tests(SubSub)
+
+    def test_two_subclasses(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+        class SubA(SomeObject):
+            pass
+
+        class SubB(SomeObject):
+            pass
+
+        self._stored_state_tests(SubA)
+        self._stored_state_tests(SubB)
```

Comment:
why wouldn't this work? :-D
added a test with it, because why not
(I had a diamond one in there before but dropped it because i thought maybe i was having too much fun)